impl_flock.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+// +build !windows,!linux
+
+package mutex
+
+import (
+	"os"
+	"syscall"
+
+	"github.com/juju/errors"
+)
+
+type impl struct {
+	fd int
+}
+
+func acquire(name string) (Releaser, error) {
+	fd, err := syscall.Open(flockName(name), syscall.O_CREAT|syscall.O_RDONLY, 0600)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	err = syscall.Flock(fd, syscall.LOCK_EX|syscall.LOCK_NB)
+	if err != nil {
+		syscall.Close(fd)
+
+		if err == syscall.EWOULDBLOCK {
+			return nil, errors.Trace(ErrLocked)
+		}
+		return nil, errors.Trace(err)
+	}
+
+	return &impl{fd: fd}, nil
+}
+
+// Release implements Releaser.
+func (i *impl) Release() error {
```

Comment:
How about this then
```go
type Releaser interface {
        // Release releases the mutex.
        // Release may be called multiple times, but only the first
        // call will release this instance of the mutex.
        // Release is unable to release the mutex successfully it
        // will call panic to forcibly release the mutex.
        Release()
}
```
Some justification

1. If Release is idempotent, ie it releases the lock then has no effect, it should be clear that calling it mulitple times won't release another instance of the _same_ lock. Ie.
```
var spec Spec = ...
r := mutex.Acquire(spec)
r.Release()
r2 := mutex.Acquire(spec)
r.Release() // does not thing, and does not affect r2
```
2. My comments about panic'ing if the mutex cannot be released are the same as elsewhere.